model/error/event_test.go

```diff
@@ -292,6 +292,114 @@ func TestErrorEventDecode(t *testing.T) {
+				map[string]interface{}{"message": "message2", "type": "type2", "parent": json.Number("0"), "cause": []interface{}{
+					map[string]interface{}{"message": "message3", "type": "type3", "cause": []interface{}{
+						map[string]interface{}{"message": "message4", "type": "type4"},
+						map[string]interface{}{"message": "message5", "type": "type5", "parent": json.Number("3")},
+					}},
+				}},
+				map[string]interface{}{"message": "message6", "type": "type6", "parent": json.Number("0")},
+			},
+		},
+	}
+	result, err := DecodeEvent(errorEvent, m.Config{}, nil)
+	require.NoError(t, err)
+
+	event := result.(*Event)
+	exceptions := flattenExceptionChain(event.Exception)
+
+	assert.Len(t, exceptions, 7)
+	for i, ex := range exceptions {
+		assert.Equal(t, fmt.Sprintf("message%d", i), *ex.Message)
+		assert.Equal(t, fmt.Sprintf("type%d", i), *ex.Type)
+		assert.Nil(t, ex.Cause)
+	}
+	assert.Equal(t, 0, *exceptions[2].Parent)
+	assert.Equal(t, 3, *exceptions[5].Parent)
+	assert.Equal(t, 0, *exceptions[6].Parent)
+}
+
+func TestDecodingAnomalies(t *testing.T) {
+
+	t.Run("exception decoder doesn't erase existing errors", func(t *testing.T) {
+		badID := map[string]interface{}{
+			"id": 7.4,
+			"exception": map[string]interface{}{
+				"message": "message0",
+				"type":    "type0",
+			},
+		}
+		result, err := DecodeEvent(badID, m.Config{}, nil)
+		assert.Error(t, err)
+		assert.Nil(t, result)
+	})
+
+	t.Run("exception decoding error bubbles up", func(t *testing.T) {
+		badException := map[string]interface{}{
+			"id": "id",
+			"exception": map[string]interface{}{
+				"message": "message0",
+				"type":    "type0",
+				"cause": []interface{}{
+					map[string]interface{}{"message": "message1", "type": 7.4},
+				},
+			},
+		}
+		result, err := DecodeEvent(badException, m.Config{}, nil)
+		assert.Error(t, err)
+		assert.Nil(t, result)
+	})
+
+	t.Run("wrong cause type", func(t *testing.T) {
+		badException := map[string]interface{}{
+			"id": "id",
+			"exception": map[string]interface{}{
+				"message": "message0",
+				"type":    "type0",
+				"cause":   []interface{}{7.4},
+			},
+		}
+		result, err := DecodeEvent(badException, m.Config{}, nil)
+		assert.Error(t, err)
+		assert.EqualError(t, err, "cause must be an exception")
+		assert.Nil(t, result)
+	})
+
+	t.Run("handle nil exceptions", func(t *testing.T) {
+		emptyCauses := map[string]interface{}{
+			"exception": map[string]interface{}{
+				"message": "message0",
+				"type":    "type0",
+				"cause": []interface{}{
+					map[string]interface{}{"message": "message1", "type": "type1", "cause": []interface{}{}},
+					map[string]interface{}{},
+				},
+			},
+		}
+		result, err := DecodeEvent(emptyCauses, m.Config{}, nil)
+		require.NoError(t, err)
+		assert.NotNil(t, result)
+
+		event := result.(*Event)
+		assert.Equal(t, *event.Exception.Message, "message0")
+		assert.Len(t, event.Exception.Cause, 1)
+		assert.Equal(t, *event.Exception.Cause[0].Message, "message1")
```

Comment:
Please ensure to have the right order on the `assert.Equal` statements, otherwise messages on test failures are confusing. Eg.  this one would lead to something like 
```
Not equal:
        	            	expected: "_whichever actual wrong message_"
        	            	actual  : "message1"
```